mockwebserver/src/test/java/okhttp3/mockwebserver/MockWebServerTest.java

```diff
@@ -432,8 +433,9 @@
     assertEquals("GET /a/deep/path?key=foo%20bar HTTP/1.1", request.getRequestLine());
 
     HttpUrl requestUrl = request.getRequestUrl();
+    String serverFQDN = InetAddress.getByName(server.getHostName()).getCanonicalHostName();
     assertEquals("http", requestUrl.scheme());
-    assertEquals(server.getHostName(), requestUrl.host());
+    assertEquals(serverFQDN, requestUrl.host());
```

Comment:
I don’t see how this could fix anything. Doesn’t requestUrl.host() just use `server.getHostName()` ? Ie. how do these end up disagreeing?

Reply:
`requestUrl.host()` is being set in RecordedRequest. The value of hostname is _localhost.localdomain_.

`String scheme = socket instanceof SSLSocket ? "https" : "http";`
`String hostname = socket.getInetAddress().getHostName();`
`int port = socket.getLocalPort();`
`this.requestUrl = HttpUrl.parse(String.format("%s://%s:%s%s", scheme, hostname, port, path));`

`socket` is coming from `MockWebServer#acceptConnections` and has the value `Socket[addr=/127.0.0.1,port=50196,localport=40867]`.

So it seems like the difference is that the host name for the MockWebServer is directly set to localhost but the RequestUrl is resolving the 127.0.0.1 to get the host name.

Reply:
For your reference, this the the contents of /etc/hosts on my machine.

```
#
# /etc/hosts: static lookup table for host names
#

#<ip-address>	<hostname.domain.org>	<hostname>
127.0.0.1	localhost.localdomain	localhost
::1		localhost.localdomain	localhost
127.0.0.1       arch.localdomain        arch

# End of file

```

Reply:
Gotcha. I think I wanna fix `MockWebServer.getHostName()` which seems to be the root cause of this problem.

How about this fix?

```
  public String getHostName() {
    before();
    return inetSocketAddress.getAddress().getCanonicalHostName();
  }
```